src/Module.Server/Common/BreakableWeaponsBehaviorServer.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Crpg.Module.Common.Network;
+using Crpg.Module.Notifications;
+using NetworkMessages.FromServer;
+using TaleWorlds.Core;
+using TaleWorlds.Engine;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.MountAndBlade.Diamond;
+using TaleWorlds.PlayerServices;
+using Timer = TaleWorlds.Core.Timer;
+
+namespace Crpg.Module.Common;
+
+internal class BreakableWeaponsBehaviorServer : MissionBehavior
+{
+    public static readonly Dictionary<string, short> BreakAbleItemsHitPoints = new()
+    {
+        { "crpg_joustinglance_striped_ry_v1_h0", 500 },
+        { "crpg_joustinglance_striped_ry_v1_h1", 550 },
+        { "crpg_joustinglance_striped_ry_v1_h2", 600 },
+        { "crpg_joustinglance_striped_ry_v1_h3", 650 },
+    };
+
+    public override MissionBehaviorType BehaviorType => MissionBehaviorType.Other;
+
+    public override void OnAgentBuild(Agent agent, Banner banner)
+    {
+        if (agent == null)
+        {
+            return;
+        }
+
+        if (agent.Equipment == null)
+        {
+            return;
+        }
+
+        for (int i = 0; i < 5; i++)
```

Comment:
The same here aswell as above:
`for (var i = EquipmentIndex.Weapon0; i <= EquipmentIndex.ExtraWeaponSlot; i += 1)`